### ISSUE TITLE
C# FindElementsByTagName doesn't work with namespace prefixed tags

### ISSUE THREAD

**Issue by csjturbine** [SeleniumHQ/selenium#7570]
## 🐛 Bug Report
The By.TagName and FindElementsByTag no longer work for namespace prefixed tags like the following: <namespace:button class="example" />

This is because the search by tag name does a css search and css does not allow the ":" character.

This is exacerbated by the inability to get the ChromeOptions.UseSpecCompliantProtocol setting to properly be set in the bindings; it now always defaults to true with ChromeDriver versions greater than v74.

see: https://bugs.chromium.org/p/chromedriver/issues/detail?id=2952 for additional detail

**Comment by jimevans** [comment]
Yes, at one point, the `tag name` locator strategy was not part of a spec compliant implementation. That was changed, and this bug remains in the .NET language bindings. It will be fixed and released in 4.0 alpha 3.

**Comment by csjturbine** [comment]
@jimevans thank you for the update.  Any idea on when that will be?  Also, are you all still shooting for Christmas 2019 for the release version of 4.0?

**Comment by jimevans** [comment]
@csjturbine The Selenium project does not publish any expected time table for any releases. There has been no (serious, non-joking) statement regarding any release date target for 4.0. The next alpha release will be available when it's ready to be consumed by users.

**Comment by csjturbine** [comment]
Again thanks for the update.  Just FYI, there are statements floating around from February of this year that mention Christmas 2019 as target date for release of 4.0 on some websites that track the Selenium project; e.g.: https://dzone.com/articles/selenium-4-is-releasing-soon-what-every-qa-must-kn

**Comment by jimevans** [comment]
@csjturbine The “release by Christmas” is a long-standing joke in the project. At the 2013 Selenium Conference in Boston, during the State of the Project address, @shs96c stated that Selenium 3.0 would be released “by Christmas.” Clearly, we didn’t meet that, with 3.0 being released two years afterward. The joke became that while we said, “before Christmas,” we never specified which year. Last summer (2018), at the Selenium Conference in India, again during the State of the Project keynote, the discussion between Simon and @manoj9788 turned to Selenium 4, and the banter revisited the same joke. Unfortunately, the background that it was a joke wasn’t clearly communicated, and wasn’t entirely understood as a joke and long-standing project lore by those in attendance. That particular article you linked to has been rebutted publicly multiple times since its publishing.

**tl;dr** Any time anyone hears that a particular release will be out “by Christmas,” it should be assumed that the reference is in jest, referring to the earlier joke.

**Comment by bjss-bristol** [comment]
I remember at one point dzone stated Christmas 2018 would be the date, which seemed to be based on this video: https://www.youtube.com/watch?v=ypmrrJmgM9U at aroundabout 43m 20s. Manjo Kumar says "this Christmas" and Simon Stewart says "a Christmas". But that didn't stop the rumour mill...

**Comment by JosephWardDotTech** [comment]
I remember that at one point dzone stated Christmas 2018 would be the date, which seemed to be based on this video: https://www.youtube.com/watch?v=ypmrrJmgM9U. At aroundabout 43m 20s. Manjo Kumar says “this Christmas” and Simon Stewart says “a Christmas”. But that didn’t stop the rumour mill...

**Comment by jimevans** [comment]
This is now a available in the latest alpha releases.
